### PR TITLE
[release-4.15] USHIFT-2810: update multiplatform builder image

### PR DESCRIPTION
**Pull request by eslutsky**

### CODE REVIEW

**Comment by eslutsky** [comment]
@thegreyd FYI

File: packaging/images/Containerfile.rpm-builder [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ARG TAG
-FROM registry.ci.openshift.org/ocp-multi/builder:$TAG
+FROM registry.ci.openshift.org/ocp/builder:$TAG
```
Comment (pacevedom) [inline packaging/images/Containerfile.rpm-builder:2]: Is this multiarch too?

Comment (eslutsky) [reply packaging/images/Containerfile.rpm-builder:2]: yes,  validated with :
```
oc image info registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.20-openshift-4.15 --filter-by-os linux/arm64
```

**Review by pacevedom — APPROVED** [review]
/lgtm

**Review by pacevedom — COMMENTED** [review]
/label cherry-pick-approved
/label backport-risk-assessed

**Comment by pacevedom** [comment]
/lgtm cancel
There is some trouble with the iamge arch

**Comment by eslutsky** [comment]
> /lgtm cancel There is some trouble with the iamge arch

@thegreyd , looks like registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.20-openshift-4.15  doesnt support ARM,
so we must use http://registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.20-builder-multi-openshift-4.16 ?

File: packaging/images/Containerfile.rpm-builder (hunk 1 above)
Comment (eslutsky) [reply packaging/images/Containerfile.rpm-builder:2]: looks like `rhel-9-golang-1.20-openshift-4.15` tag doesnt with arm, so we must use `rhel-9-golang-1.20-builder-multi-openshift-4.16` which is the same  image but  mutli arch.

**Comment by pacevedom** [comment]
/rename [release-4.15] USHIFT-2810: update multiplatform builder image

**Comment by pacevedom** [comment]
/retitle [release-4.15] [USHIFT-2810](https://issues.redhat.com//browse/USHIFT-2810): update multiplatform builder image

**Review by pacevedom — COMMENTED** [review]
/lgtm

**Comment by pacevedom** [comment]
/retitle [release-4.15] USHIFT-2810: update multiplatform builder image

File: Makefile [hunk 2]
```diff
@@ -112,7 +112,7 @@ GO_TEST_PACKAGES=./cmd/... ./pkg/...
 export CGO_ENABLED ?= 1
 
 # Specify OCP build tools image tag when building rpm with podman
-RPM_BUILDER_IMAGE_TAG := rhel-9-golang-latest-openshift-4.15
+RPM_BUILDER_IMAGE_TAG := rhel-9-golang-1.20-builder-multi-openshift-4.16
```
Comment (ggiguash) [inline Makefile:115]: We're in 4.15 branch. Why are we using a 4.16 package?

Comment (ggiguash) [inline Makefile:115]: Please, add an explanation in the comment on why we're using 4.16 package.

**Pull request merged**